concrete/src/Page/Sitemap/Element/SitemapPage.php

```diff
@@ -0,0 +1,307 @@
+use DateTime;
+use SimpleXMLElement;
+
+class SitemapPage extends SitemapElement
+{
+    /**
+     * The page.
+     *
+     * @var \Concrete\Core\Page\Page
+     */
+    protected $page;
+
+    /**
+     * The URL of the page.
+     *
+     * @var \Concrete\Core\Url\UrlInterface
+     */
+    protected $url;
+
+    /**
+     * The last modification date/time.
+     *
+     * @var \DateTime|null
+     */
+    protected $lastMod;
+
+    /**
+     * The page change frequency.
+     *
+     * @var string
+     */
+    protected $changeFrequency;
+
+    /**
+     * The page priority.
+     *
+     * @var string
+     */
+    protected $priority;
+
+    /**
+     * The pages in alternative languages mapped to this page.
+     *
+     * @var \Concrete\Core\Page\Sitemap\Element\SitemapPageAlternativeLanguage[]
+     */
+    protected $alternativeLanguages = [];
+
+    /**
+     * Should this page be skipped?
+     *
+     * @var bool
+     */
+    protected $skip = false;
+
+    /**
+     * @param Page $page the page
+     * @param UrlInterface $url the URL of the page
+     * @param \DateTime|null the last modification date/time
+     * @param string $changeFrequency the page change frequency
+     * @param string $priority the page priority
+     * @param null|DateTime $lastMod
+     */
+    public function __construct(Page $page, UrlInterface $url, DateTime $lastMod = null, $changeFrequency = '', $priority = '')
+    {
+        $this->page = $page;
+        $this->setUrl($url);
+        $this->setLastMod($lastMod);
+        $this->setChangeFrequency($changeFrequency);
+        $this->setPriority($priority);
+    }
+
+    /**
+     * Get the page.
+     *
+     * @return \Concrete\Core\Page\Page
+     */
+    public function getPage()
+    {
+        return $this->page;
+    }
+
+    /**
+     * Get the URL of the page.
+     *
+     * @return \Concrete\Core\Url\UrlInterface
+     */
+    public function getUrl()
+    {
+        return $this->url;
+    }
+
+    /**
+     * Set the URL of the page.
+     *
+     * @param \Concrete\Core\Url\UrlInterface $url
+     *
+     * @return $this
+     */
+    public function setUrl(UrlInterface $url)
+    {
+        $this->url = $url;
+
+        return $this;
+    }
+
+    /**
+     * Get the last modification date/time.
+     *
+     * @return \DateTime|null
+     */
+    public function getLastMod()
+    {
+        return $this->lastMod;
+    }
+
+    /**
+     * Set the last modification date/time.
+     *
+     * @param \DateTime|null $lastMod
+     *
+     * @return $this
+     */
+    public function setLastMod(DateTime $lastMod = null)
+    {
+        $this->lastMod = $lastMod;
+
+        return $this;
+    }
+
+    /**
+     * Get the page change frequency.
+     *
+     * @return string
+     */
+    public function getChangeFrequency()
+    {
+        return $this->changeFrequency;
+    }
+
+    /**
+     * Set the page change frequency.
+     *
+     * @param string $changeFrequency
+     *
```

Comment:
I think an example here would help understand how this works. Or maybe it's better to explain that at the propertly level? E.g. what kind of values can be in `$changeFrequency`? (same for `priority`)

Reply:
:+1: I added constants to enumerate the valid values